iroha/src/block_sync.rs

```diff
@@ -0,0 +1,180 @@
+use self::message::*;
+use crate::{
+    block::ValidBlock,
+    kura::Kura,
+    peer::PeerId,
+    sumeragi::{InitializedNetworkTopology, Role, Sumeragi},
+};
+use async_std::{sync::RwLock, task};
+use std::{sync::Arc, time::Duration};
+
+/// The state of `BlockSync`.
+#[derive(Clone)]
+pub enum State {
+    /// Not synchronizing now.
+    Idle,
+    /// Requested blocks from other peers.
+    Requested,
+    /// Synchronization is in progress: validating and committing blocks.
+    InProgress(Vec<ValidBlock>),
```

Comment:
Variants naming may be better. Requested variant is good because we have such naming for transactions. Idle raises question - is it waiting to be processed for the first time, or it turned back from processing to idle? Would be great to add Lifecycle into documentation.
InProgress reminds me about Telecom with eTOMS lifecycle, but we have not such variants for transactions so maybe we can wrote a better term?

Reply:
Okay, I will think on naming and provide a diagram in a separate RFC.